src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
It's not problematic for my work, as I explained I worked around it, however it *will* be problematic for extensions that do not have access to cmark's private headers, and I would suggest two things if the API is to stay that way:

* Document this new constraint
* Expose API to let extensions query the memory allocator for a given node or parser